src/System.Linq.Expressions/tests/TestExtensions/PerCompilationTypeAttribute.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+using Xunit.Sdk;
+
+namespace System.Linq.Expressions.Tests
+{
+    /// <summary>
+    /// Operates as per <see cref="MemberDataAttribute"/>, but adds a final boolean value to the list of arguments,
+    /// permuted through both false and true.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    internal class PerCompilationTypeAttribute : DataAttribute
```

Comment:
That'd certainly be more useful generally. and the boxing could be reduced with some caching. Since this is going to be applicable throughout these tests, and the more general case won't, I think I'll leave it.